client/src/components/course-results-view/CourseResultsTable.js

```diff
@@ -0,0 +1,198 @@
+              attainments = {attainments}
+            />
+            <TableBody>
+              {stableSort(studentsToShow,
+                getComparator(order, orderBy))
+                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                .map((student) => {
+
+                  return (
+                    <TableRow
+                      hover
+                      tabIndex={-1}
+                      key={student.studentID}
+                    >
+                      <TableCell
+                        sx={{ width: '100px' }}
+                        component="th"
+                        id={student.studentID}
+                        scope="row"
+                        padding="normal"
+                      >
+                        {student.studentID}
+                      </TableCell>
+                      {attainments.map((attainment) => {
+                        return (
+                          <TableCell
+                            sx={{ width: '100px' }}
+                            align="left"
+                            key={`${student.studentID}_${attainment.id}`}>
+                            {student[attainment.id]}
+                          </TableCell>
+                        );
+                      })}
+                      <TableCell
+                        sx={{ width: '100px' }}
+                        align="left"
+                        key={`${student.studentID}_finalGrade`}>
+                        {student.finalGrade}
+                      </TableCell>
+                    </TableRow>
+                  );
+                })}
+              {emptyRows > 0 && (
+                <TableRow
+                  style={{
+                    height: (dense ? 33 : 53) * emptyRows,
+                  }}
+                >
+                  <TableCell colSpan={6} />
+                </TableRow>
+              )}
+            </TableBody>
+          </Table>
+        </TableContainer>
+        <Box sx={{ 
+          display: 'flex', 
+          flexDirection: 'row',
+          flexWrap: 'wrap',
+          justifyContent: 'space-between',
+          py: '10px'
+        }}>
+          <Box sx={{ display: 'flex', flexDirection: 'row', ml: 3.5 }}>
+            <Typography sx={{ mt: '11px' }} >View valid grades from past instances:</Typography>
+            <Button sx={{ ml: 1, mt: '10px', height:'30px' }} size='small'>View all grades</Button>
+          </Box>
+          <TablePagination
+            rowsPerPageOptions={[25, 50, 100, 500]}
+            component="div"
+            count={studentsToShow.length}
+            rowsPerPage={rowsPerPage}
+            page={page}
+            onPageChange={handleChangePage}
+            onRowsPerPageChange={handleChangeRowsPerPage}
+          />
+        </Box>
+      </Paper>
+      <FormControlLabel
+        control={<Switch checked={dense} onChange={handleChangeDense} />}
+        label="Dense padding"
```

Comment:
Love this!

Reply:
Modified the final grade calculation so that it sends an error message if no student grades have been added -> now the mock data also works as intended